server/audit_math/suite.py

```diff
@@ -1,718 +0,0 @@
-
-        Outputs:
-            pvalue: the pvalue from testing the hypothesis that null margin is not the acual margin
-        """
-
-        if self.sample_size == 0 or reported_margin == 0:
-            return 1.0
-
-        if self.sample_size == self.num_ballots:
-            return 0.0
-
-        sample = bravo.compute_cumulative_sample(self.sample)
-        n_w = sample[winner]
-        n_l = sample[loser]
-        n_u = self.sample_size - n_w - n_l
-
-        v_w = self.vote_totals[winner]
-        v_l = self.vote_totals[loser]
-        v_u = self.num_ballots - v_w - v_l
-
-        null_margin = (v_w - v_l) - null_lambda * reported_margin
-
-        if not (v_w >= n_w and v_l >= n_l and v_u >= n_u):
-            raise ValueError("Alternative hypothesis isn't consistent with the sample")
-
-        alt_logLR = (
-            np.sum(np.log(v_w - np.arange(n_w)))
-            + np.sum(np.log(v_l - np.arange(n_l)))
-            + np.sum(np.log(v_u - np.arange(n_u)))
-        )
-
-        null_logLR = (
-            lambda Nw: (n_w > 0) * np.sum(np.log(Nw - np.arange(n_w)))
-            + (n_l > 0) * np.sum(np.log(Nw - null_margin - np.arange(n_l)))
-            + (n_u > 0)
-            * np.sum(np.log(self.num_ballots - 2 * Nw + null_margin - np.arange(n_u)))
-        )
-
-        upper_n_w_limit = (self.num_ballots - n_u + null_margin) / 2.0
-        lower_n_w_limit = np.max([n_w, n_l + null_margin])
-
-        # For extremely small or large null_margins, the limits do not
-        # make sense with the sample values.
-        if upper_n_w_limit < n_w or (upper_n_w_limit - null_margin) < n_l:
-            raise ValueError("Null is impossible, given the sample")
-
-        LR_derivative = (
-            lambda Nw: np.sum([1 / (Nw - i) for i in range(n_w)])
-            + np.sum([1 / (Nw - null_margin - i) for i in range(n_l)])
-            - 2
-            * np.sum(
-                [1 / (self.num_ballots - 2 * Nw + null_margin - i) for i in range(n_u)]
-            )
-        )
-
-        # Check if the maximum occurs at an endpoint: deriv has no sign change
-        if LR_derivative(upper_n_w_limit) * LR_derivative(lower_n_w_limit) > 0:
-            nuisance_param = (
-                upper_n_w_limit
-                if null_logLR(upper_n_w_limit) >= null_logLR(lower_n_w_limit)
-                else lower_n_w_limit
-            )
-        # Otherwise, find the (unique) root of the derivative of the log likelihood ratio
-        else:
-            nuisance_param = sp.optimize.brentq(
-                LR_derivative, lower_n_w_limit, upper_n_w_limit
-            )
-        logLR = alt_logLR - null_logLR(nuisance_param)
-        LR = float(np.exp(logLR))  # This value is always a float, but np.exp
-        # can return a vector. casting for the typechecker.
-        # Note if this value overflows, the p-value becomes 0.
-
-        return 1.0 / LR if 1.0 / LR < 1 else 1.0
```

Comment:
nit: why not `min(1.0/LR, 1.0)` like they have in the original code?

Reply:
that returns `Optional` which causes mypy to freak out

Reply:
I just tried it and mypy didn't complain. Maybe the issue was using `np.amin` vs just the builtin `min`?

Reply:
Oh maybe so. I can update it.